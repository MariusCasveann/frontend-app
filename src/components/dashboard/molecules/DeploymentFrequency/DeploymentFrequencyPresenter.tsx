import React from 'react';
import { Card } from '../../../common/molecules/CardLayout/CardLayout';
import './DeploymentFrequency.css';
import DeploymentFrequencyWidget from './DeploymentFrequencyWidget';
import { FiltersContext } from '../../atoms/Filters/FiltersContext';

export default () => (
    <FiltersContext.Consumer>
        {({ fromDate, toDate, financialYear, messageWhenNoData }) => (
            <Card className="deployment-frequency-container" title="Deployment Frequency KPIs" width={100}>
                <DeploymentFrequencyWidget
                    messageWhenNoData={messageWhenNoData}
                    fromDate={fromDate}
                    toDate={toDate}
                    financialYear={financialYear}
                />
            </Card>
        )}
    </FiltersContext.Consumer>
);
