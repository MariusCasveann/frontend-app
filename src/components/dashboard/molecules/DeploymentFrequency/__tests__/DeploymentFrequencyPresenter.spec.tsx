import * as React from 'react';
import renderer from 'react-test-renderer';
jest.mock('../../../molecules/GoogleCost/GoogleCostPresenter', () => 'GoogleCostPresenter');
jest.mock('../DeploymentFrequencyWidget', () => 'DeploymentFrequencyWidget');
jest.doMock('../../../atoms/Filters/FiltersContext', () => ({
    FiltersContext: 'FiltersContext',
    'FiltersContext.Consumer': 'FiltersContext.Consumer'
}));
import DeploymentFrequencyPresenter from '../DeploymentFrequencyPresenter';

describe(' DeploymentFrequencyPresenter', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<DeploymentFrequencyPresenter />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
