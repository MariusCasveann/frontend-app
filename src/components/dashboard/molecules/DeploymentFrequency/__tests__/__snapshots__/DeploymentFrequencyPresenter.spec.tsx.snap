// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` DeploymentFrequencyPresenter renders correctly 1`] = `
<div
  className="deployment-frequency-container card"
  style={
    Object {
      "flex": "1 1 100%",
    }
  }
>
  <div
    className="card-title-wrapper"
  >
    <div
      className="card-title"
    >
      Deployment Frequency KPIs
    </div>
  </div>
  <div
    className="card-content-data"
  >
    <div
      className="card-content-wrapper"
    >
      <DeploymentFrequencyWidget
        financialYear={null}
        fromDate="2019-01-01"
        messageWhenNoData="No data available, please select another time and/or domain/product."
        toDate="2019-01-02"
      />
    </div>
  </div>
</div>
`;
